### ISSUE TITLE
Example for securely setting secrets for something like Google Analytics

### ISSUE THREAD

**Issue by hamirmahal** [aklinker1/vite-plugin-web-extension#226]
### Summary

It'd be nice if there were an example that securely sets secrets for a service, like Google Analytics, for example.

### Additional Context

Since some Chrome extension authors will likely be using it anyway, Google Analytics could be a good use case for this.

**Comment by aklinker1** [comment]
There is no way to securely store secrets in chrome extensions. Some way or another, the key is always shipped in your JS code.

Double check to see if your key is actually a secret. If it's not, and it's more of a public key, just put it in your `.env` file. If it's a actual secret that shouldn't be leaked, check the permissions. For the Google Analytics Measurement Protocol, while the API key is a secret, all it can do is report events, it can't change anything... so I include it in my `.env` file as well.

If it's a true secret that should never be shipped in your codebase, don't put it in your codebase. Use an external API and use the secret server-side to make external requests. An example of this would be a OpenAI API key.

Alternatively, if you're not comfortable shipping a key as-is in your code, you could use a 2-way encryption algorithm to ship the encoded string rather than your real string. But you're gonna have to ship your algorithm to decode it, meaning an attacker can still get it if they want.

TLDR: there are no secrets in frontend web dev.

**Comment by aklinker1** [comment]
I don't provide any documentation around environment variables because this is a Vite plugin, so it uses the same environment variable setup as Vite.

https://vite.dev/guide/env-and-mode.html

**Comment by hamirmahal** [comment]
Thanks for the response.

Would it be feasible, at the very least, to ensure a request is coming from an extension environment?

**Comment by hamirmahal** [comment]
I'm thinking about going with this approach like you mentioned.

> use the secret server-side to make external requests

I suppose an attacker could still spam the server-side endpoint with bogus analytics events from external environments.
